runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -1097,9 +1097,32 @@ void generateReportFieldAccessOutlinedInstructions(TR::Node *node, TR::LabelSymb
       {
       TR::DataType dt = node->getDataType();
       int32_t elementSize = dt == TR::Address ? TR::Compiler->om.sizeofReferenceField() : TR::Symbol::convertTypeToSize(dt);
-      TR::InstOpCode::Mnemonic storeOp = getLoadOrStoreFromDataType(cg, dt, elementSize, node->getOpCode().isUnsigned(), false);
+      TR::InstOpCode::Mnemonic storeOp = TR::InstOpCode::bad;
+      uint32_t valueMRSize = 0;
+
+      if (TR::Compiler->om.compressObjectReferences() && TR::Address == dt && comp->target().is64Bit())
```

Comment:
i still felt something might be wrong here.  for example, do you distinguish the following two aspects:
1) read vs. write;
2) static field vs. instance field;

static field doesn't contain compressed value (?) i believed. furthermore, instance field might not be of type TR:Address to begin with, since loading instance field looks like  iloadi, then i2l, then l2a.

Reply:
Read vs Write is handled on line 1096 where it checks `isWrite`.

The data type is coming from the `awrtbar`/`awrtbari` which has type `TR:Address`. Will that sometimes be a `iwrtbar`/`iwrtbari` when writing a compressed value? If so, how can I tell if it is writing an address or not?

The static vs instance handling might be wrong since I missed that case. I'll try confirming what it should be then fix it.

Reply:
you need to put more thoughts into this fix.  have you evaluated the trade-off between decompressing the value in the caller of this routine vs. in this routine itself?  for example.